torch_xla/csrc/xla_lower_util.cpp

```diff
@@ -247,4 +248,62 @@ xla::XlaOp CreateIndex(const xla::XlaOp& input, const xla::XlaOp& indices) {
   return xla::Gather(input, indices, dim_numbers, slice_sizes);
 }
 
+xla::XlaOp CreateIndexUpdate(
+    const xla::XlaOp& buffer, const xla::XlaOp& indices,
+    const xla::XlaOp& values,
+    const std::function<xla::XlaOp(xla::XlaOp, xla::XlaOp, xla::XlaBuilder*)>&
+        combiner) {
+  const xla::Shape buffer_shape = XlaHelpers::ShapeOfXlaOp(buffer);
+  const xla::Shape indices_shape = XlaHelpers::ShapeOfXlaOp(indices);
+  xla::Shape values_shape = XlaHelpers::ShapeOfXlaOp(values);
+
+  absl::Span<const xla::int64> indices_dims =
+      xla::AsInt64Slice(indices_shape.dimensions());
+  // The minor dimension of indices contains the indices to update.
+  xla::int64 num_index_dims = 1;
+  XLA_CHECK(!indices_dims.empty());
+  num_index_dims = indices_dims.back();
+  indices_dims.remove_suffix(1);
+  xla::ScatterDimensionNumbers dim_numbers;
+  dim_numbers.set_index_vector_dim(indices_shape.dimensions_size() - 1);
+
+  xla::int64 values_rank = values_shape.rank();
+  xla::int64 buffer_rank = buffer_shape.rank();
+  xla::int64 num_window_dims_in_values = buffer_rank - num_index_dims;
+
+  // Make the values match the rank expected by scatter.
+  std::vector<xla::int64> expected_values_dims(indices_dims.begin(),
+                                               indices_dims.end());
+  for (xla::int64 dim = num_index_dims; dim < buffer_rank; ++dim) {
+    expected_values_dims.push_back(buffer_shape.dimensions(dim));
+  }
+  xla::XlaOp new_values = BuildExpand(values, expected_values_dims);
+  values_shape = XlaHelpers::ShapeOfXlaOp(new_values);
+  values_rank = values_shape.rank();
+
+  for (xla::int64 i = (values_rank - num_window_dims_in_values);
+       i < values_rank; ++i) {
+    dim_numbers.add_update_window_dims(i);
+  }
+  for (xla::int64 i = 0; i < num_index_dims; ++i) {
+    dim_numbers.add_inserted_window_dims(i);
+    dim_numbers.add_scatter_dims_to_operand_dims(i);
+  }
+  // Build the combiner computation.
+  xla::XlaComputation combiner_computation;
+  {
```

Comment:
I don't think there is the need for this fake scope.

Reply:
Done.